### PR TITLE
Fix relative paths resolution inside babelrc

### PR DESCRIPTION
**Pull request by fix-fix**
Resolve babel presets and plugins paths against current directory.

Previously `.babelrc` config file was parsed and the result was passed to `babel.transformFile`, causing presets and plugins relative paths to be resolved against each transformed file, and not `.babelrc` itself.

---

This change is crucial for using `.babelrc.js` instead of `.babelrc` in pre babel 7.0 applications (by specifying `"presets": ["./.babelrc.js"]`, [more info](https://github.com/babel/babel/pull/4892#issuecomment-304537510)). And since babel-plugin-react-intl doesn't support 7.0 yet, it makes sense to support this workaround in other react-intl tools supporting babel 6.x.

As far as I can tell this change is backward-compatible (I tested it with different babel configs and envs), but unfortunately I didn't manage to write tests to prove that.

### CODE REVIEW

**Comment by fix-fix** [comment]
@akameco looks like all tests on Node 6 fail because of `async` keyword in test suite.

**Comment by akameco** [comment]
@fix-fix Thank you 👍 

I fixed test on Node 6. Please rebase master.

File: index.js [hunk 1]
```diff
@@ -20,10 +20,21 @@ const concatArray = (obj, src) => {
...
+const resolvePresets = (presets, cwd) =>
+  presets.map(
+    preset => (isString(preset) ? resolvePreset(preset, cwd) : preset)
+  )
+
+const resolvePlugins = (plugins, cwd) =>
+  plugins.map(
+    plugin => (isString(plugin) ? resolvePlugin(plugin, cwd) : plugin)
+  )
+
```
Comment (akameco) [inline index.js]: Duplicate.
I like the abstraction like `createResolveList`

```js
const createResolveList = fn => (list, cwd) =>
  list.map(x => (typeof x === 'string' ? fn(x, cwd) : x))

const resolvePresets = createResolveList(resolvePreset)
const resolvePlugins = createResolveList(resolvePlugin)
```

**Review by akameco — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (fix-fix) [reply index.js]: I rebased against master and applied your suggestion

**Review by akameco — APPROVED** [review]

**Comment by akameco** [comment]
@fix-fix Thank you 💯

**Comment by fix-fix** [comment]
Thanks for merging. When can the new release of `extract-react-intl-messages` be expected?

In `extract-react-intl-messages`'s package.json the specified range for `extract-react-intl` is `^0.4.0`, which doesn't match new version 0.5.0, thus fixed version can't be used yet.

**Comment by akameco** [comment]
@fix-fix Oh! Sorry! I will release it soon.

**Pull request merged**